docarray/typing/url/any_url.py

```diff
@@ -38,6 +52,29 @@ def _to_node_protobuf(self) -> 'NodeProto':
 
         return NodeProto(text=str(self), type=self._proto_type_name)
 
+    @classmethod
+    def is_extension_allowed(cls, value: Any) -> bool:
+        """
+        Check if the file extension of the URL is allowed for this class.
+        First, it guesses the mime type of the file. If it fails to detect the
+        mime type, it then checks the extra file extensions.
+
+        :param value: The URL or file path.
+        :return: True if the extension is allowed, False otherwise
+        """
+        if cls is AnyUrl:
+            return True
+
+        url_parts = value.split("?")
+        mimetype, _ = mimetypes.guess_type(url_parts[0])
+        if mimetype and mimetype.startswith(cls.mime_type()):
+            return True
+
+        filename = url_parts[0].split('.')
+        extension = filename[-1] if len(filename) > 1 else None
```

Comment:
maybe I am being overly cautious here, but do we know for a fact that there are no corner cases where this splitting into filename and extension could break? Is there some resource or standard that we can reference?
Alternatively, I think pydantic implements some of this internally, Maybe we could repurpose some of their logic?

Reply:
but I think this is ok for now, wdyt?